apps/sch/fsw/unit_test/sch_app_test.c

```diff
@@ -1016,199 +983,240 @@ void SCH_TimerInit_Test_CustomEarlyInitError(void)
-    SCH_AppData.UnexpectedMajorFrameCount   = 10;
-    SCH_AppData.MissedMajorFrameCount       = 11;
-    SCH_AppData.ValidMajorFrameCount        = 12;
-    SCH_AppData.WorstCaseSlotsPerMinorFrame = 13;
-    SCH_AppData.ClockAccuracy = 14;
+    /* The following will emulate the behavior of receiving a message,
+       and gives it data to process. */
+    ADPipe = Ut_CFE_SB_CreatePipe(SCH_AD_PIPE_NAME);
+    CFE_SB_InitMsg((void*)&DoneMsg, SCH_UNUSED_MID, sizeof(DoneMsg), TRUE);
+    Ut_CFE_SB_AddMsgToPipe((void*)&DoneMsg, (CFE_SB_PipeId_t)ADPipe);
 
-    /* Set to satisfy subsequent condition "Status != CFE_SUCCESS" */
-    Ut_OSAPI_SetReturnCode(UT_OSAPI_BINSEMCREATE_INDEX, -1, 1);
+    Ut_CFE_ES_SetReturnCode(UT_CFE_ES_REGISTERCHILDTASK_INDEX,
+                            CFE_SUCCESS, 1);
 
-    /* Execute the function being tested */
-    Result = SCH_TimerInit();
-    
-    /* Verify results */
-    UtAssert_True(SCH_AppData.IgnoreMajorFrame     == FALSE, "SCH_AppData.IgnoreMajorFrame     == FALSE");
-    UtAssert_True(SCH_AppData.IgnoreMajorFrameMsgSent == FALSE, "SCH_AppData.IgnoreMajorFrameMsgSent == FALSE");
-    UtAssert_True(SCH_AppData.UnexpectedMajorFrame == FALSE, "SCH_AppData.UnexpectedMajorFrame == FALSE");
-    UtAssert_True(SCH_AppData.SyncToMET            == SCH_NOT_SYNCHRONIZED, "SCH_AppData.SyncToMET            == SCH_NOT_SYNCHRONIZED");
-    UtAssert_True(SCH_AppData.MajorFrameSource     == SCH_MAJOR_FS_NONE, "SCH_AppData.MajorFrameSource     == SCH_MAJOR_FS_NONE");
-    UtAssert_True(SCH_AppData.NextSlotNumber       == 0, "SCH_AppData.NextSlotNumber       == 0");
-    UtAssert_True(SCH_AppData.MinorFramesSinceTone == SCH_TIME_SYNC_SLOT, "SCH_AppData.MinorFramesSinceTone == SCH_TIME_SYNC_SLOT");
-    UtAssert_True(SCH_AppData.LastSyncMETSlot      == 0, "SCH_AppData.LastSyncMETSlot      == 0");
-    UtAssert_True(SCH_AppData.SyncAttemptsLeft     == 0, "SCH_AppData.SyncAttemptsLeft     == 0");
-    UtAssert_True(SCH_AppData.UnexpectedMajorFrameCount   == 0, "SCH_AppData.UnexpectedMajorFrameCount   == 0");
-    UtAssert_True(SCH_AppData.MissedMajorFrameCount       == 0, "SCH_AppData.MissedMajorFrameCount       == 0");
-    UtAssert_True(SCH_AppData.ValidMajorFrameCount        == 0, "SCH_AppData.ValidMajorFrameCount        == 0");
-    UtAssert_True(SCH_AppData.WorstCaseSlotsPerMinorFrame == 1, "SCH_AppData.WorstCaseSlotsPerMinorFrame == 1");
-    UtAssert_True(SCH_AppData.ClockAccuracy = SCH_WORST_CLOCK_ACCURACY, "SCH_AppData.ClockAccuracy = SCH_WORST_CLOCK_ACCURACY");
+    /* To finish the Child Task after receiving a message */
+    Ut_CFE_SB_SetReturnCode(UT_CFE_SB_RCVMSG_INDEX, CFE_SB_PIPE_RD_ERR, 2);
 
-    UtAssert_True
-        (Ut_CFE_EVS_EventSent(SCH_SEM_CREATE_ERR_EID, CFE_EVS_ERROR, "Error creating Main Loop Timing Semaphore (RC=0xFFFFFFFF)"),
-        "Error creating Main Loop Timing Semaphore (RC=0xFFFFFFFF)");
+    /* Execute the function being tested */
+    SCH_ADChildTask();
```

Comment:
In case the ticket #283 is resolved, then the other two cases works but the one that I added RcvMsg hook function wouldn't work.
So I changed back to the code that didn't have RcvMsg Hook.
All three test cases will work when the ticket #283 is resolved.